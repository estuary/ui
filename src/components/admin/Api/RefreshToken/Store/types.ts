import { PostgrestError } from '@supabase/postgrest-js';

export interface RefreshTokenState {
    token: string;
    setToken: (value: string) => void;

    description: string;
    updateDescription: (value: string) => void;

    saving: boolean;
    setSaving: (value: boolean) => void;

    serverError: PostgrestError | null;
    setServerError: (value: PostgrestError | string | null) => void;

    resetState: () => void;
}
