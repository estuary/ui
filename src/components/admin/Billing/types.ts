export interface AdminBillingProps {
    showAddPayment?: boolean;
}
