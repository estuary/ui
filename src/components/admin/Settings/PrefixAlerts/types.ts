export interface EmailDictionary {
    [prefix: string]: string[];
}
