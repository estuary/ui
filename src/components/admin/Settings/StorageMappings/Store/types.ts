import type { PostgrestError } from '@supabase/postgrest-js';
import type { CloudProviderCodes } from 'src/components/admin/Settings/StorageMappings/Dialog/useConfigurationSchema';
import type { JsonFormsData } from 'src/types';

export interface StorageMappingState {
    provider: CloudProviderCodes | null;
    updateProvider: (value: CloudProviderCodes) => void;

    formValue: JsonFormsData;
    updateFormValue: (value: JsonFormsData) => void;

    pubId: string;
    setPubId: (value: string) => void;

    logToken: string;
    setLogToken: (value: string) => void;

    saving: boolean;
    setSaving: (value: boolean) => void;

    serverError: PostgrestError | null;
    setServerError: (value: PostgrestError | string | null) => void;

    resetState: () => void;
}
