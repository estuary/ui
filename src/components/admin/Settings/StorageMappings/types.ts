import type { DataPlaneOption } from 'src/stores/DetailsForm/types';

export interface DataPlaneSelectorProps {
    options: DataPlaneOption[];
}
