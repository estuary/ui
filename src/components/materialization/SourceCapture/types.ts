export interface SourceCaptureOptionInfoProps {
    messageKey: string;
}
