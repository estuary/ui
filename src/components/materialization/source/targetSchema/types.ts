import type { BaseAutoCompleteOption } from 'src/components/shared/specPropEditor/types';
import type { TargetSchemas } from 'src/stores/SourceCapture/types';

export interface AutoCompleteOption extends BaseAutoCompleteOption {
    val: TargetSchemas;
}
