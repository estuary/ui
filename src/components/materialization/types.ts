export interface TrialOnlyPrefixAlertProps {
    messageId: string;
    triggered: boolean;
    bindingUUID?: string;
}
