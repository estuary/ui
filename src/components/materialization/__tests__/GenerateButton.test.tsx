import { render } from '@testing-library/react';
import { AuthenticatedOnlyContext } from 'context/Authenticated';
import { EntityContextProvider } from 'context/EntityContext';
import { renderOps } from 'test/test-utils';
import { Entity } from 'types';
import MaterializeGenerateButton from '../GenerateButton';

const renderComponent = (entityType: Entity) => {
    return render(
        <AuthenticatedOnlyContext>
            <EntityContextProvider value={entityType}>
                <MaterializeGenerateButton disabled={false} />
            </EntityContextProvider>
        </AuthenticatedOnlyContext>,
        renderOps
    );
};

describe('MaterializeGenerateButton', () => {
    test('should render', async () => {
        const rendered = renderComponent('materialization');

        expect(rendered.baseElement).toMatchSnapshot();
    });
});
