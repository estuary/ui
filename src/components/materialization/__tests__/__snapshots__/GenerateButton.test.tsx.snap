// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MaterializeGenerateButton > should render 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-outlinedError MuiAlert-outlined css-11js1a5-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          color="currentColor"
          fill="none"
          font-size="14px"
          height="1.5em"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="1.5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.172 14.828L12.001 12m2.828-2.828L12.001 12m0 0L9.172 9.172M12.001 12l2.828 2.828M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
      >
        <div
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-o8x25m-MuiTypography-root-MuiAlertTitle-root"
        >
          Error
        </div>
        There was an unexpected application error. 
        Expand to see details.
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <button
          aria-expanded="false"
          aria-label="show more"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1r502bh-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            color="currentColor"
            fill="none"
            font-size="14px"
            height="1.5em"
            stroke-width="1.5"
            style="font-size: 14px; color: rgb(11, 19, 30);"
            viewBox="0 0 24 24"
            width="1.5em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 9l6 6 6-6"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
      </div>
    </div>
  </div>
</body>
`;
